watchdog/watcher/manager.go

```diff
@@ -25,78 +25,107 @@ import (
 
 type Manager interface {
 	Close()
-	Get(rsName string) *Watcher
-	HasWatcher(rsName string) bool
-	Stop(rsName string)
-	Watch(rs *replset.Replset)
+	Get(serviceName, rsName string) *Watcher
+	HasWatcher(serviceName, rsName string) bool
+	Stop(serviceName, rsName string)
+	Watch(serviceName string, rs *replset.Replset)
+}
+
+type watcherState struct {
+	rsName      string
+	serviceName string
+	watcher     *Watcher
+	quit        chan bool
 }
 
 type WatcherManager struct {
 	sync.Mutex
 	config     *config.Config
 	stop       *chan bool
-	quitChans  map[string]chan bool
-	watchers   map[string]*Watcher
+	watchers   []*watcherState
```

Comment:
what about replace the slice with the map that has key `rsName: serviceName`. It it will simplify lookup operations